src/Tuura/SelfDual.hs

```diff
@@ -0,0 +1,43 @@
+module Tuura.SelfDual (
+    isSelfDual, getSelfDuals) where
+
+import Control.Monad (replicateM)
+import Data.Either (isRight)
+import Data.List (sort)
+
+import Tuura.Boolean
+
+isSelfDual :: DNF String -> Bool
```

Comment:
It seems this can be generalised to `Ord a => DNF a -> Bool`, which would be a more useful function.